### PR TITLE
DidHelper Remove Verification Method

### PR DESCRIPTION
**Pull request by abdulmth**
Fixes https://github.com/KILTprotocol/ticket/issues/3368

### CODE REVIEW

File: packages/sdk-js/src/DidHelpers/setVerificationMethod.ts [hunk 1]
```diff
@@ -82,3 +84,82 @@ export function setVerificationMethod(
...
   })
 }
+
+/**
+ * Removes the verification method for the selected `verificationMethodId` and  `relationship`.
+ *
+ * Note: authentication verification method can not be removed.
+ *
+ * @param options Any {@link SharedArguments} and additional parameters.
+ * @param options.relationship The relationship for which this verification method shall be useable.
```
Comment (rflechtner) [inline packages/sdk-js/src/DidHelpers/setVerificationMethod.ts:94]: It's the opposite, right? The relationship for which this VM shall be deactivated / become unusable.

File: packages/sdk-js/src/DidHelpers/setVerificationMethod.ts [hunk 2]
```diff
@@ -82,3 +84,82 @@ export function setVerificationMethod(
...
+  switch (options.relationship) {
+    case 'authentication': {
+      throw new Error('authentication verification methods can not be removed')
+    }
+    case 'capabilityDelegation': {
+      if (
+        options.didDocument.capabilityDelegation &&
+        options.didDocument.capabilityDelegation.includes(
+          options.verificationMethodId
+        )
```
Comment (rflechtner) [inline packages/sdk-js/src/DidHelpers/setVerificationMethod.ts:115]:
```suggestion
        options.didDocument.capabilityDelegation?.includes(
          options.verificationMethodId
        )
```
Does this work or will TS complain?

**Review by rflechtner — APPROVED** [review]
LGTM! I'd personally like to have a sign-off by @Dudleyneedham & @ntn-x2 given that removing verification methods was not part of the finalised and agreed-upon solution doc (https://hackmd.io/YEKFloxtR36v5w1ODppszA?view), but I think we have a workable solution here. Once VM management becomes more flexible, we can start making some parameters optional (e.g., the `relationship` parameter).

File: packages/sdk-js/src/DidHelpers/setVerificationMethod.ts [hunk 3]
```diff
@@ -82,3 +84,79 @@ export function setVerificationMethod(
     expectedEvents: [{ section: 'did', method: 'DidUpdated' }],
   })
 }
+
+/**
+ * Removes the verification method for the selected `verificationMethodId` and  `relationship`.
```
Comment (Dudleyneedham) [inline packages/sdk-js/src/DidHelpers/setVerificationMethod.ts:89]:
```suggestion
 * Removes the verification method for the selected `verificationMethodId` and `relationship`.
```

**Review by Dudleyneedham — APPROVED** [review]
LGTM

**Comment by rflechtner** [comment]
I guess this can be closed now, given that it's been merged with https://github.com/KILTprotocol/sdk-js/pull/880?

**Review by ntn-x2 — COMMENTED** [review]
Apparently I'm too late to approve 👍

**Pull request closed without merging**